pkg/globalnet/controllers/pod_controller.go

```diff
@@ -0,0 +1,108 @@
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package controllers
+
+import (
+	"github.com/submariner-io/admiral/pkg/federate"
+	"github.com/submariner-io/admiral/pkg/syncer"
+	"github.com/submariner-io/admiral/pkg/util"
+	submarinerv1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog"
+)
+
+func startPodController(svc *corev1.Service, config syncer.ResourceSyncerConfig) (*podController, error) {
+	var err error
+
+	_, gvr, err := util.ToUnstructuredResource(&corev1.Pod{}, config.RestMapper)
+	if err != nil {
+		return nil, err
+	}
+
+	controller := &podController{
+		baseSyncerController: newBaseSyncerController(),
+		pods:                 config.SourceClient.Resource(*gvr),
+		scheme:               config.Scheme,
+		svcName:              svc.Name,
+		namespace:            svc.Namespace,
+		ingressIPMap:         make(map[string]bool),
+	}
+
+	labelSelector := labels.Set(svc.Spec.Selector).AsSelector().String()
+
+	controller.resourceSyncer, err = syncer.NewResourceSyncer(&syncer.ResourceSyncerConfig{
+		Name:                "Pod syncer",
+		ResourceType:        &corev1.Pod{},
+		SourceClient:        config.SourceClient,
+		SourceNamespace:     svc.Namespace,
+		RestMapper:          config.RestMapper,
+		Federator:           federate.NewCreateFederator(config.SourceClient, config.RestMapper, corev1.NamespaceAll),
+		Scheme:              config.Scheme,
+		Transform:           controller.process,
+		SourceLabelSelector: labelSelector,
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	if err := controller.Start(); err != nil {
+		return nil, err
+	}
+
+	klog.Infof("Creating Pod controller for (%s/%s) with selector %q", svc.Namespace, svc.Name, labelSelector)
+
+	return controller, nil
+}
+
+func (c *podController) process(from runtime.Object, numRequeues int, op syncer.Operation) (runtime.Object, bool) {
+	pod := from.(*corev1.Pod)
+
+	ingressIP := &submarinerv1.GlobalIngressIP{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      GetIngressIPName(pod),
+			Namespace: pod.Namespace,
+		},
+	}
+
+	if op == syncer.Delete {
+		delete(c.ingressIPMap, pod.Name)
+		klog.Infof("Delete Pod %s/%s", pod.Namespace, pod.Name)
+
+		return ingressIP, false
+	}
+
+	if c.ingressIPMap[pod.Name] || pod.Status.Phase != corev1.PodRunning || pod.Status.PodIP == "" {
```

Comment:
What is the intent of the `ingressIPMap` check here?

Reply:
This is to check if we have already created globalIngressIP for this pod. A pod may get further updates but we're not interested if we already have assigned it a globalIngressIP

Reply:
We can ignore pod updates - you're using the create-only federator anyway as there's nothing to update on the `GlobalIngressIP`.